### PR TITLE
Add a copy capability to jupyter notebooks

### PR DESCRIPTION
**Pull request by agrippa**
This is a small change to add the ability to copy files in the Notebook UI, in addition to moving them. Because of my infamiliarity with the notebook code base, I tried to minimize the amount of changes necessary to achieve this. Mostly, this appropriates the same code paths as Move but adds a keep_old flag that indicates whether the source file should be cleared out (move) or not (copy).

This is my first pull request on the repo, so apologies if I skipped/mangled a whole bunch of steps in creating this pull request (as I imagine I may have).

### CODE REVIEW

**Comment by takluyver** [comment]
Thanks! It will probably be a few days before I can look at this properly, but someone else might be able to have a look sooner.

It looks like the CI tests have failed - can you have a look and see if that's related to the change you made?

There is already a method `ContentsManager.copy()` in the API - maybe it's possible to use that?

File: notebook/static/base/js/namespace.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ define(function(){
     // tree
     jglobal('SessionList','tree/js/sessionlist');
 
-    Jupyter.version = "5.2.1";
+    Jupyter.version = "5.3.0.dev0";
```
Comment (dibyadas) [inline notebook/static/base/js/namespace.js]: @agrippa I think version changes automatically when start developing the code base. And should be changed when we are making a PR, if I'm not mistaken.
I'm currently new to the codebase so I might be wrong in this.  😄

**Comment by takluyver** [comment]
`ContentsManager` already has a `copy()` method - can we try using that rather than adding a parameter to `rename()`? Or is there some reason that won't work?

**Comment by agrippa** [comment]
@takluyver Thanks for the feedback! I looked at just using copy() and did a bit of refactoring, is that along the lines  of what you were thinking?

**Comment by takluyver** [comment]
That's a start, but I don't think we need the `keep_old` parameter anywhere - the contents manager class already has a `copy()` method separate from `rename()`. Maybe we need to expose it to the notebook UI.

**Comment by Zsailer** [comment]
Thanks, @agrippa, for this work!

This has been inactive for awhile, so I'm going to close this PR for now. 

I think it's a good feature to add at some point, but it looks like developer time is limited here. If anyone has time to devote to it, feel free reopen or resubmit a new PR. 

Thanks!

**Pull request closed without merging**